### PR TITLE
user mode

### PR DESCRIPTION
**Pull request by pgajdos**

### CODE REVIEW

File: src/lib/fonts/fonts-config-state.rb [hunk 1]
```diff
@@ -269,12 +269,40 @@ class FontsConfigState
...
+        if (File.exists?(@agent_file_path))
+          SCR.RegisterAgent(path(@agent_path),
+                            term(:ag_ini, term(:SysConfigFile, @agent_file_path)))
+        end
+      end
+    end
+
+    def finalize_agent
+      if (!@root_user)
+        SCR.UnregisterAgent(path(".userconfig.fonts-config"))
```
Comment (jreidinger) [inline src/lib/fonts/fonts-config-state.rb]: why not use @agent_path here?

File: src/lib/fonts/fonts-config-state.rb [hunk 2]
```diff
@@ -318,56 +346,63 @@ def to_s
    end
 
    def write
+      if (!@root_user && !File.exists?(@agent_file_path))
+        mkdir_p(File.dirname(@agent_file_path))
+        touch(@agent_file_path)
+      end
```
Comment (jreidinger) [inline src/lib/fonts/fonts-config-state.rb]: maybe add this to initialize agent itself?

File: src/lib/fonts/fonts-dialog.rb [hunk 3]
```diff
@@ -298,8 +300,8 @@ def handle_presets_button(widget, event)
         if CWMTab.CurrentTab == "specimens"
           initialize_specimen_widget("")
         elsif CWMTab.CurrentTab == "algorithms"
-          initialize_aaoff_checkbox("")
-          initialize_aamonooff_checkbox("")
+          initialize_antialias_checkbox("")
+          initialize_antialias_mono_checkbox("")
```
Comment (jreidinger) [inline src/lib/fonts/fonts-dialog.rb]: maybe use default param with `""` so in this case you call only `initialize_antialias_mono_checkbox`

File: src/lib/fonts/fonts-dialog.rb [hunk 4]
```diff
@@ -952,10 +954,13 @@ def run_dialog
...
                                   "tabs_fonts_configuration"),
-          "caption"            => _("Font Configuration"),
+          "caption"            => _("Font Configuration") +  
+                                  (root_user? ? "" : _(" (User Mode)")),
           "next_button"        => Label.OKButton,
           "abort_button"       => Label.AbortButton,
-          "back_button"        => "",
+          # misuse back_button a bit
+          "back_button"        => (root_user? || !File.exists?(FontsConfigCommand.user_sysconfig_file) ? 
+                                     nil : _("&Use system settings")),
```
Comment (jreidinger) [inline src/lib/fonts/fonts-dialog.rb]: nice usage :+1:

File: src/lib/fonts/shell-commands.rb [hunk 5]
```diff
@@ -54,7 +59,7 @@ def self.fonts_config_file(file_id)
 
       cmd = "#{FONTS_CONFIG_CMD} --info"
       result = Yast::SCR.Execute(BASH_SCR_PATH, cmd)
-      file = result["stdout"].lines.select{|l| l =~ /#{file_id}/}[0].gsub(/.*: /, '').gsub(/\n/, '')
+      file = result["stdout"].lines.select{|l| l =~ /^[ ]*#{file_id}:/}[0].gsub(/.*: /, '').gsub(/\n/, '')
```
Comment (jreidinger) [inline src/lib/fonts/shell-commands.rb]: if you want filter out whispaces, better way is `\s*` see e.g. for playing http://rubular.com/

**Comment by jreidinger** [comment]
few ideas for improvements otherwise it looks really good.

**Comment by pgajdos** [comment]
Thanks, all suggestions implemented.

**Comment by jreidinger** [comment]
One more note I have after night thinking :)
https://github.com/yast/yast-fonts/blob/master/src/desktop/fonts.desktop#L11 is still set to true, which means that this module do not show in control center for non-root users, but after your change I think it make sense to have it there.

**Pull request merged**